### ISSUE TITLE
Parsing of formatted code failed

### ISSUE THREAD

**Issue by waddlaw** [tweag/ormolu#128]
```hs
f = -- comment
  case () of
    () -> ()
```

```shell
λ ormolu Bug.hs
Parsing of formatted code failed:
<rendered>:2:1
parse error (possibly incorrect indentation or mismatched brackets)
Please, consider reporting the bug.
```

It seems to occur when `case` and `--` are combined.

**Comment by mitchellwrosen** [comment]
Another failure:

```
module Foo where

foo =
  do
    x
  +
  do
    y
```

**Comment by basile-henry** [comment]
Another parse error:

```haskell
f x | x = 42
```

```sh
Parsing of formatted code failed:
<rendered>:1:7
parse error on input ‘|’
Please, consider reporting the bug.
```

**Comment by mrkkrp** [comment]
This one is harder and should be fixed as part of the changes described in #94 and #95. I'll probably get to do this next week.

**Comment by mrkkrp** [comment]
This

```
f = -- comment
  case () of
    () -> ()
```

seems to be resolved as a side-effect of #156.

**Comment by mrkkrp** [comment]
Going to close this one because it looks like a kitchen sink to various unrelated issues. The only one that is still not fixed is [this one](https://github.com/tweag/ormolu/issues/128#issuecomment-504614712) and it looks like a duplicate of #176 to me.
